helix-term/src/commands/typed.rs

```diff
@@ -1801,22 +1800,48 @@ fn toggle_option(
     let pointer = format!("/{}", key.replace('.', "/"));
     let value = config.pointer_mut(&pointer).ok_or_else(key_error)?;
 
-    let Value::Bool(old_value) = *value else {
-        anyhow::bail!("Key `{}` is not toggle-able", key)
+    *value = match value.as_bool() {
+        Some(value) => serde_json::Value::Bool(!value),
+        None => {
+            if args.len() != 2 {
+                anyhow::bail!(
+                    "Bad arguments. For non-boolean configurations use: `:toggle key val1,val2`"
+                );
+            }
+            if !value.is_string() {
+                anyhow::bail!("Bad configuration. Cannot cycle non-string configurations");
+            }
```

Comment:
you can use the `ensure` macro for checks like this

Reply:
Neat, see https://github.com/helix-editor/helix/pull/4411/commits/998426efcf754b738914a9d92b04e1a6a5bca5aa